src/scripts/main.js

```diff
@@ -1,3 +1,172 @@
+          <option>New York</option>
+          <option>Edinburgh</option>
+          <option>San Francisco</option>
+        </select>
+      </label>
+      <label>Age: <input name="age" type="number" data-qa="age"></label>
+      <label>Salary: <input name="salary" type="number" data-qa="salary"></label>
+      <button type="button" id="save-button">Save to table</button>
+    </form>
+  `;
+
+  document.body.insertAdjacentHTML('beforeend', formHtml);
+
+  const form = document.querySelector('.new-employee-form');
+  const saveButton = document.getElementById('save-button');
+
+  saveButton.addEventListener('click', () => {
+    const employeeName = form.name.value.trim();
+    const position = form.position.value.trim();
+    const office = form.office.value;
+    const age = form.age.value;
+    const salary = form.salary.value;
+
+    if (employeeName.length < 4) {
+      showNotification(
+        'Error',
+        'Name must be at least 4 characters long',
+        'error',
+      );
+
+      return;
+    }
+
+    if (age < 18 || age > 90) {
+      showNotification('Error', 'Age must be between 18 and 90', 'error');
+
+      return;
+    }
+
+    if (!employeeName || !position || !office || !age || !salary) {
+      showNotification('Error', 'All fields are required', 'error');
+
+      return;
+    }
+
+    const formattedSalary = `$${Number(salary).toLocaleString('en-US')}`;
+
+    const newRow = document.createElement('tr');
+
+    newRow.innerHTML = `
+      <td>${employeeName}</td>
+      <td>${position}</td>
+      <td>${office}</td>
+      <td>${age}</td>
+      <td>${formattedSalary}</td>
+    `;
+    employeeTable.querySelector('tbody').appendChild(newRow);
+
+    showNotification('Success', 'Employee added to the table', 'success');
+    form.reset();
+  });
+
+  function showNotification(title, message, type) {
+    const notification = document.createElement('div');
+
+    notification.className = `notification ${type}`;
+    notification.setAttribute('data-qa', 'notification');
+
+    notification.innerHTML = `
+      <span class="title">${title}</span>
+      <p>${message}</p>
+    `;
+    document.body.appendChild(notification);
+
+    setTimeout(() => {
+      notification.remove();
+    }, 3000);
+  }
+
+  employeeTable.addEventListener('dblclick', function (e) {
+    const cell = e.target;
+
+    if (cell.tagName === 'TD') {
+      const initialText = cell.innerText;
+      const input = document.createElement('input');
+
+      input.className = 'cell-input';
+      input.value = initialText;
+      cell.innerText = '';
+      cell.appendChild(input);
+      input.focus();
+
+      input.addEventListener('blur', () => {
+        const newValue = input.value.trim();
+
+        cell.innerText = newValue === '' ? initialText : newValue;
+      });
+
+      input.addEventListener('keypress', (evt) => {
+        if (evt.key === 'Enter') {
+          input.blur();
+        }
+      });
+    }
+  });
```

Comment:
The logic for editing a cell value on double click is a bit complex. Consider breaking it down into smaller functions to improve readability and maintainability of your code.